go-controller/pkg/libovsdb/util/acl.go

```diff
@@ -239,16 +240,29 @@ func convertK8sProtocolToOVNProtocol(proto v1.Protocol) string {
 type NetworkPolicyPort struct {
 	Protocol string // will store the OVN protocol string syntax for the corresponding K8s protocol
 	Port     int32  // will store startPort if its a range
-	EndPort  int32
+	EndPort  int32  // will store 0 if its not a range
+	// saves NamedPort representation;
+	// key is the name of the Port
+	// value is an array of possible representations of this port (relevance wrt to rule, peers)
+	NamedPort map[string]*[]NamedNetworkPolicyPort
```

Comment:
> I am also not sure why it's a map, seems like it always has max 1 entry with port name?

right its always 1 entry only goal was to save the string name and then the list of ports there..

Reply:
> I think maybe it should be a separate struct. Because named port is not really a port, it is a set of ip+port pairs, so they should be used differently e.g. when generating an acl match. It also seems like there are a lot of `ifs` in current functions to handle them separately

hmm good point I already have the NamedNetworkPolicyPort, I could do a separate representation for this and not club this with the existing struct.

Reply:
done! made it an independent entity and it makes much more sense and code is cleaner.